lib/ddtrace/contrib/rails/action_controller.rb

```diff
@@ -15,6 +15,10 @@ def self.instrument
           end
         end
 
+        def self.custom_span_tags
```

Comment:
I really like the idea of being able to set  custom span tags from within controller.

However using class variables to do that will probably cause quite a bit of problems especially when running from within multithreaded servers like Puma.

Looking at existing implementation there seems to be no easy way to do it.

One thing that comes to my mind is to expose span from this [object](https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/rails/core_extensions.rb#L209) like so:

```ruby
module ActionControllerPatch
   def datadog_current_span
     Datadog::Contrib::Rails::ActionController.current_span(@datatog_current_payload) if @datatog_current_payload
   end
  
   def process_action_with_datadog(*args)
   # ...
     Datadog::Contrib::Rails::ActionController.start_processing(payload)
     @datadog_current_payload = payload
     result = super(*args) 
   # ...
   end
end

#...

module Datadog
  module Contrib
    module Rails
      module ActionController
      #...
        def self.current_span(payload)
          tracing_context = payload.fetch(:tracing_context)
          tracing_context[:dd_request_span] if tracing_context
        end
     end
  end
end
#...
```

This should expose `span` object from within controller, while also allow us to swap implementation to something else in the future. 

/cc: @delner

Reply:
sounds really good to me, to have payload defined/extended outside, please feel free to make any required changes to this PR!